### PR TITLE
Add some verbose logging to bitcoin-util-test.py

### PR DESCRIPTION
**Pull request by jnewbery**
Adds a new command line parameter to output verbose logging when running `bitcoin-util-test.py` manually.

### CODE REVIEW

**Comment by jnewbery** [comment]
Sample output:

```
~/bitcoin/src$ test/bitcoin-util-test.py --srcdir='.' --verbose
Testing: Creates a blank transaction
PASS
Testing: Creates a blank transaction (output in json)
PASS
Testing: Creates a blank transaction when nothing is piped into bitcoin-tx
PASS
Testing: Creates a blank transaction when nothing is piped into bitcoin-tx (output in json)
PASS
Testing: Deletes a single input from a transaction
PASS
Testing: Deletes a single input from a transaction (output in json)
PASS
Testing: Attempts to delete an input with a bad index from a transaction. Expected to fail.
PASS
Testing: Deletes a single output from a transaction
PASS
Testing: Deletes a single output from a transaction (output in json)
PASS
Testing: Attempts to delete an output with a bad index from a transaction. Expected to fail.
PASS
Testing: Adds an nlocktime to a transaction
PASS
Testing: Adds an nlocktime to a transaction (output in json)
PASS
Testing: Creates a new transaction with three inputs and two outputs
PASS
Testing: Creates a new transaction with three inputs and two outputs (output in json)
PASS
Testing: Creates a new transaction with a single empty output script
PASS
Testing: Creates a new transaction with a single empty output script (output in json)
PASS
Testing: Creates a new transaction with a single input and a single output, and then signs the transaction
PASS
Testing: Creates a new transaction with a single input and a single output, and then signs the transaction (output in json)
PASS
Testing: Attempts to create a new transaction with one input and an output with malformed hex data. Expected to fail
PASS
Testing: Attempts to create a new transaction with one input and an output with no value and malformed hex data. Expected to fail
PASS
Testing: Creates a new transaction with one input, one address output and one data output
PASS
Testing: Creates a new transaction with one input, one address output and one data output. Output in json
PASS
Testing: Creates a new transaction with one input, one address output and one data (zero value) output
PASS
Testing: Creates a new transaction with one input, one address output and one data (zero value) output. Output in json
PASS
Testing: Creates a new transaction with one input with sequence number and one address output
PASS
Testing: Creates a new transaction with one input with sequence number and one address output. Output in json
PASS
Testing: Adds a new input with sequence number to a transaction
PASS
Testing: Adds a new input with sequence number to a transaction. Output in json
PASS
```

File: src/test/data/bitcoin-util-test.json [hunk 1]
```diff
@@ -136,15 +157,17 @@
      "in=5897de6bd6027a475eadd57019d4e6872c396d0716c4875a5f1a6fcfdf385c1f:0",
      "outaddr=0.18:13tuJJDR2RgArmgfv6JScSdreahzgc4T6o",
      "outdata=4:54686973204f505f52455455524e207472616e73616374696f6e206f7574707574207761732063726561746564206279206d6f646966696564206372656174657261777472616e73616374696f6e2e"],
-    "output_cmp": "txcreatedata1.json"
+    "output_cmp": "txcreatedata1.json",
+    "description": "Creates a new transaction with one input, one address output and one data output. Output in json"
```
Comment (fanquake) [inline src/test/data/bitcoin-util-test.json]: nit: lines from here down can be changed to use "(output in json)" to match the above.

**Comment by fanquake** [comment]
Getting the same output. etc
```
bash-3.2$ test/bitcoin-util-test.py --srcdir='.' --verbose
Testing: Creates a blank transaction
PASS
Testing: Creates a blank transaction (output in json)
PASS
.......
Testing: Adds a new input with sequence number to a transaction
PASS
Testing: Adds a new input with sequence number to a transaction. Output in json
PASS
```

**Comment by jnewbery** [comment]
Thanks @fanquake . Descriptions fixed and commits squashed.

**Comment by MarcoFalke** [comment]
utACK 621441a7a72ec0b17594a24a2c5fedcac0c605f6

**Comment by jonasschnelli** [comment]
utACK 621441a7a72ec0b17594a24a2c5fedcac0c605f6

**Pull request merged**